### PR TITLE
feat(runtime): construct pint runtimes for kusama and polkadot

### LINKED ISSUE #156: extract common config types to a common crate
## Issue summary

in preparation of multiple runtime configurations we should extract commonly used types to a `common` crate within `runtime`, this may include
- common runtime parameter types (RuntimeBlockWeights, ...)
- balance unit types an helper functions (unit, microunit, milliunit...)
- anything we can precompile for multiple runtimes

## Other information and links

-

### LINKED ISSUE #240: Construct PINT runtimes for polkadot, kusama and dev(rococo)
## Issue summary

- [ ] pint-polkadot
- [ ] pint-kusama
- [ ] pint-dev

## Other information and links

- https://github.com/ChainSafe/PINT/issues/156#issuecomment-887319313
- #156 
- #159 
- #236

### LINKED ISSUE #242: Add crate `node/client` for conditional compilation
## Issue summary

https://github.com/paritytech/polkadot/tree/master/node/client

## Other information and links
- #240

### PR DESCRIPTION
**Pull request by clearloop**
## Changes

- rename `/runtime` to `/runtime/dev`
- add `/runtime/kusama`
- add `/runtime/polkadot`
- conditional compilation in `/node`
- gather dup configs in `/runtime/common`
- move types back to `/primitives`
- [x] make node compile again
- [x] remove versbose configurations

## Tests

```
cargo t --all --all-features
```

## Issues

- Closes #242 
- Closes #240
- Closes #156

### CODE REVIEW

File: node/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,12 @@
 // Copyright 2021 ChainSafe Systems
 // SPDX-License-Identifier: LGPL-3.0-only
 
+#[cfg(not(any(feature = "kusama", feature = "polkadot")))]
+pub use pint_runtime_dev as pint_runtime;
+#[cfg(feature = "kusama")]
+pub use pint_runtime_kusama as pint_runtime;
+#[cfg(feature = "polkadot")]
+pub use pint_runtime_polkadot as pint_runtime;
```
Comment (clearloop) [inline node/src/lib.rs:9]: `kusama` and `polkadot` are derived from `dev`, use `dev` runtime by default.

**Review by mattsse — APPROVED** [review]
LGTM

File: node/src/lib.rs (hunk 1 above)
Comment (mattsse) [reply node/src/lib.rs:9]: I guess that's reasonable for now, but as a next step we should create separate services for each rt

**Review by mattsse — CHANGES_REQUESTED** [review]
but needs a fix of the duplicate import error in `node`

File: runtime/dev/src/lib.rs [hunk 2]
```diff
@@ -508,15 +406,6 @@ impl pallet_saft_registry::Config for Runtime {
     type WeightInfo = weights::pallet_saft_registry::WeightInfo<Runtime>;
 }
 
-parameter_types! {
-    pub const ProposalSubmissionPeriod: <Runtime as frame_system::Config>::BlockNumber = 10;
-    pub const VotingPeriod: <Runtime as frame_system::Config>::BlockNumber = 10;
-}
```
Comment (mattsse) [inline runtime/dev/src/lib.rs:514]: I'd like to make this network dependent, so that we can for example use different thresholds for the dev runtime

**Review by mattsse — CHANGES_REQUESTED** [review]
Awesome! this looks great already, I've some nits and suggestions on how to organise some parts of the code.

File: runtime/common/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+pub type BlockNumber = u32;
+
+/// Digest item type.
+pub type DigestItem = generic::DigestItem<Hash>;
+
+/// Identifier for price feeds.
+pub type FeedId = u64;
+
+/// A hash of some data used by the chain.
+pub type Hash = sp_core::H256;
```
[the remark this answers is not in the document]
Comment (clearloop) [reply runtime/common/src/types.rs:40]: Agree with this, both Acala and Polkadot writing these types in `primitives`

**Comment by clearloop** [comment]
> Awesome! this looks great already, I've some nits and suggestions on how to organize some parts of the code.

Still WIP...more complex than expected, hard to build with `--all-features` unless refactor the `client.rs` and `service.rs`

File: runtime/dev/src/lib.rs [hunk 4]
```diff
@@ -124,39 +112,14 @@ pub fn native_version() -> NativeVersion {
...
-/// We allow for 0.5 seconds of compute with a 6 second average block time.
-const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND / 2;
-
 parameter_types! {
-    pub const BlockHashCount: BlockNumber = 250;
+    pub Ancestry: MultiLocation = Junction::Parachain(
+        ParachainInfo::parachain_id().into()
+    ).into();
+    pub RelayChainOrigin: Origin = cumulus_pallet_xcm::Origin::Relay.into();
+    pub const RelayNetwork: NetworkId = NetworkId::Polkadot;
```
[the remark this answers is not in the document]
Comment (clearloop) [reply runtime/dev/src/lib.rs:120]: - `Ancestry` depends on pallet `ParachainInfo`
- `Version` should be independent as well

**Review by mattsse — APPROVED** [review]
LGTM

**Pull request merged**